report_substitute/static/src/js/action_manager.js

```diff
@@ -0,0 +1,37 @@
+// Copyright 2019 ACSONE SA/NV
+// License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+odoo.define("report_substitute.action_report_substitute", function (require) {
+    "use strict";
+
+    var ActionManager = require("web.ActionManager");
+
+    ActionManager.include({
+
+        /**
+         * Intercept action handling substitute the report action
+         * @override
+         */
+
+        _handleAction: function (action, options) {
+            if (action.type === "ir.actions.report" &&
+                action.context.active_ids) {
```

Comment:
Is there some indication in action that there are subsitution rules for this report? To save a server round-trip is there are no substitution rules.